### PR TITLE
[RHCLOUD-24563] Set restrictive env config as default value

### PR DESCRIPTION
**Pull request by g-duval**
Use flags in a consistent way by making the restrictive mode the default

### CODE REVIEW

**Review by MikelAlejoBR — APPROVED** [review]
LGTM!

**Review by gwenneg — CHANGES_REQUESTED** [review]
Do not merge (yet), I'll take a look later.

File: .rhcicd/clowdapp-engine.yaml [hunk 1]
```diff
@@ -327,7 +329,7 @@ parameters:
   value: "false"
 - name: NOTIFICATIONS_EMAILS_ONLY_MODE_ENABLED
```
Comment (gwenneg) [inline .rhcicd/clowdapp-engine.yaml:330]: `clowdapp-backend.yaml` also contains that flag. Both files should have the same flag value.

File: .rhcicd/clowdapp-engine.yaml [hunk 2]
```diff
@@ -438,3 +440,5 @@ parameters:
   value: ""
 - name: NOTIFICATIONS_USE_SECURED_EMAIL_TEMPLATES_ENABLED
   value: "true"
+- name: NOTIFICATIONS_USE_TEMPLATES_FROM_DB
+  value: "true"
```
Comment (gwenneg) [inline .rhcicd/clowdapp-engine.yaml:444]: Is that really needed in this PR?

Comment (gwenneg) [reply .rhcicd/clowdapp-engine.yaml:444]: It is, I just remembered why 😄

File: .rhcicd/clowdapp-engine.yaml (hunk 1 above)
Comment (g-duval) [reply .rhcicd/clowdapp-engine.yaml:330]: Right, I forgot this :)

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**